auth-web/src/components/auth/manage-business/BusinessLookup.vue

```diff
@@ -96,61 +131,82 @@ enum States {
 /*
  * See PPR's BusinessSearchAutocomplete.vue for a Composition API example.
  */
-@Component({})
-export default class BusinessLookup extends Vue {
-  // enum for template
-  readonly States = States
-
-  // action from business module
-  @BusinessModule.Action('isAffiliated')
-  private readonly isAffiliated!: (identifier: string) => Promise<boolean>
-
-  /** V-model for search field. */
-  searchField = ''
-
-  /** Results from business lookup API. */
-  searchResults = [] as Array<BusinessLookupResultIF>
-
-  /** State of this component. */
-  state = States.INITIAL
-
-  /** Called when searchField property has changed. */
-  @Watch('searchField')
-  @Debounce(600)
-  private async onSearchFieldChanged (): Promise<void> {
-    // safety check
-    if (this.searchField && this.searchField?.length < 3) {
-      this.state = States.TYPING
-    } else if (this.searchField && this.searchField?.length > 2) {
-      this.state = States.SEARCHING
-      const searchStatus = null // search all (ACTIVE + HISTORICAL)
-      this.searchResults = await BusinessLookupServices.search(this.searchField, searchStatus).catch(() => [])
-
-      // enable or disable items according to whether they have already been added
-      for (const result of this.searchResults) {
-        result.disabled = await this.isAffiliated(result.identifier)
+// @Component({})
+export default defineComponent({
+  name: 'BusinessLookup',
+  props: {
+    lookupType: { type: String, default: LookupType.BUSINESS }
+  },
+  setup (props, { emit }) {
```

Comment:
I think usually we use `emits: ['']`

Reply:
I checked other components, they used { emit } as well, type error if change to emits: ['']

Reply:
You need to use both, otherwise you get a warning when you emit

Reply:
![image](https://github.com/bcgov/sbc-auth/assets/3484109/7f6279c1-a5b9-434c-ad6b-f668c81c7546)

Reply:
https://vuejs.org/guide/components/events.html#declaring-emitted-events

Reply:
![image](https://github.com/bcgov/sbc-auth/assets/3484109/45659dd0-b06e-4d48-ae49-4baf1946015d)

Reply:
updated